src/riscv/lib/src/jit/builder/control_flow_graph.rs

```diff
@@ -85,14 +86,17 @@ pub struct StepCounterUpdate<'info, T> {
     /// Unsuccessful outcomes increase by just `base_diff`.
     base_diff: usize,
 
+    /// If the program exits after this outcome, the step counter must be increased by this amount.
```

Comment:
Add why this needs to be done. I.e. because the subsequent step counter updates won't fire.